hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeRecordReaderUtils.java

```diff
@@ -189,7 +190,13 @@ public static Writable avroToArrayWritable(Object value, Schema schema) {
         Writable[] recordValues = new Writable[schema.getFields().size()];
         int recordValueIndex = 0;
         for (Schema.Field field : schema.getFields()) {
-          recordValues[recordValueIndex++] = avroToArrayWritable(record.get(field.name()), field.schema());
+          Object fieldValue = null;
+          try {
+            fieldValue = record.get(field.name());
+          } catch (AvroRuntimeException e) {
+            LOG.debug("Field:" + field.name() + "not found in Schema:" + schema.toString());
```

Comment:
I remember we have performance concerns regarding to this change. Catching exceptions is not efficient.

Reply:
The`GenericRecord` Api does not have the `hasField` method in avro 1.8.2 https://avro.apache.org/docs/1.8.2/api/java/org/apache/avro/generic/GenericRecord.html#hasField-java.lang.String-
so i think in general when it performs this get we have to do some exception catching i believe. 

its only present in the 1.10.2 https://avro.apache.org/docs/1.10.2/api/java/org/apache/avro/generic/GenericRecord.html#hasField-java.lang.String-

Reply:
I think also the perforance concerns are not with the actual try catch but more so to do with the exception catching https://www.oreilly.com/library/view/programming-jakarta-struts/0596006519/ch10s02.html#:~:text=In%20general%2C%20wrapping%20your%20Java,proper%20handler%20for%20the%20exception. 

```
In general, wrapping your Java code with try/catch blocks doesn’t have a significant performance impact on your applications. Only when exceptions actually occur is there a negative performance impact, which is due to the lookup the JVM must perform to locate the proper handler for the exception.
```

In this case though i think we will need to catch the exception